doc/classes/EditorPlugin.xml

```diff
@@ -292,6 +292,35 @@
 				[/codeblocks]
 			</description>
 		</method>
+		<method name="forward_spatial_draw_over_viewport" qualifiers="virtual">
+			<return type="void">
+			</return>
+			<argument index="0" name="overlay" type="Control">
+			</argument>
+			<description>
+				Called by the engine when the 3D editor's viewport is updated. Use the [code]overlay[/code] [Control] for drawing. You can update the viewport manually by calling [method update_overlays].
+				[codeblock]
+				func forward_spatial_draw_over_viewport(overlay):
+					#draw a circle at cursor position
+					overlay.draw_circle(overlay.get_local_mouse_position(), 64)
+
+				func forward_spatial_gui_input(camera, event):
+					if event is InputEventMouseMotion:
+						#redraw viewport when cursor is moved
+						update_overlays()
```

Comment:
This should return a boolean wheter or not the event is consumend

Reply:
I just noticed. But is it really necessary? The method will return `null` by default (I think) and adding `true` doesn't change much, as it's motion event.

Reply:
Yeah, but it is required in C# and it will generate an inconsistency.